### DISCUSSION TITLE
Add citations from external packages / functions (liionpack)

### DISCUSSION THREAD

**Discussion by TomTranter** [pybamm-team/PyBaMM#2857] [General]
Would it be a good idea to be able to add external citations for functions not in PyBaMM but used as part of simulation workflow? 
My specific use case is liionpack where I am using PyBaMM but the functions I use are not part of the PyBaMM core package so it doesn't fit the use case to add an entry to CITATIONS.txt and then register it in liionpack.
Solutions could be:

1. Replicate the functionality of the citation registration in liionpack by subclassing it and adding the extra reference manually or
2. We could add the ability to register from an external citations file or simply a string or dict with all the citation info so that pybamm.print_citations() still works and users can add their own in dependent packages.

**Comment by brosaplanella** [comment]
I think the second would be nicer as then we can easily use it with any future packages we (or other people) might want to develop. I guess this means we need to figure out how to deal with clashes (i.e. two different papers with the same key).
